orchagent/bfdorch.cpp

```diff
@@ -0,0 +1,412 @@
+};
+
+const map<sai_bfd_session_state_t, string> session_state_loopup =
+{
+    {SAI_BFD_SESSION_STATE_ADMIN_DOWN,  "Admin_Down"},
+    {SAI_BFD_SESSION_STATE_DOWN,        "Down"},
+    {SAI_BFD_SESSION_STATE_INIT,        "Init"},
+    {SAI_BFD_SESSION_STATE_UP,          "Up"}
+};
+
+BfdOrch::BfdOrch(DBConnector *db, string tableName):
+    Orch(db, tableName)
+{
+    SWSS_LOG_ENTER();
+
+    m_state_db = shared_ptr<DBConnector>(new DBConnector("STATE_DB", 0));
+    m_stateBfdSessionTable = unique_ptr<Table>(new Table(m_state_db.get(), STATE_BFD_SESSION_TABLE));
+
+    DBConnector *notificationsDb = new DBConnector("ASIC_DB", 0);
+    m_bfdStateNotificationConsumer = new swss::NotificationConsumer(notificationsDb, "NOTIFICATIONS");
+    auto bfdStateNotificatier = new Notifier(m_bfdStateNotificationConsumer, this, "BFD_STATE_NOTIFICATIONS");
+    Orch::addExecutor(bfdStateNotificatier);
+}
+
+BfdOrch::~BfdOrch(void)
+{
+    SWSS_LOG_ENTER();
+}
+
+void BfdOrch::doTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string key =  kfvKey(t);
+        string op = kfvOp(t);
+        auto data = kfvFieldsValues(t);
+
+        if (op == SET_COMMAND)
+        {
+            if (!create_bfd_session(key, data))
+            {
+                it++;
+                continue;
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (!remove_bfd_session(key))
+            {
+                it++;
+                continue;
+            }
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s\n", op.c_str());
+        }
+
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+void BfdOrch::doTask(NotificationConsumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    std::string op;
+    std::string data;
+    std::vector<swss::FieldValueTuple> values;
+
+    consumer.pop(op, data, values);
+
+    if (&consumer != m_bfdStateNotificationConsumer)
+    {
+        return;
+    }
+
+    if (op == "bfd_session_state_change")
+    {
+        uint32_t count;
+        sai_bfd_session_state_notification_t *bfdSessionState = nullptr;
+
+        sai_deserialize_bfd_session_state_ntf(data, count, &bfdSessionState);
+
+        for (uint32_t i = 0; i < count; i++)
+        {
+            sai_object_id_t id = bfdSessionState[i].bfd_session_id;
+            sai_bfd_session_state_t state = bfdSessionState[i].session_state;
+
+            SWSS_LOG_INFO("Get bfd session state change notification id:%" PRIx64 " state:%d", id, state);
+
+            if (state != bfd_session_lookup[id].state)
+            {
+                auto key = bfd_session_lookup[id].peer;
+                m_stateBfdSessionTable->hset(key, "state", session_state_loopup.at(state));
+
+                bfd_session_lookup[id].state = state;
```

Comment:
Can you please add the code for notifying observers? similar to port oper change?

Reply:
Added code for notifying observer. For this PR, there will not be anywhere processing these notifications though.